tests/test_frame.py

```diff
@@ -180,9 +179,8 @@ class MyEnum(Enum):
 
 
 def test_slice_setitem() -> None:
-    # Due to the bug in pandas 1.2.3(https://github.com/pandas-dev/pandas/issues/40440), this is in separate test case
     df = pd.DataFrame(data={"col1": [1, 2], "col2": [3, 4], 5: [6, 7]})
-    df[1:] = ["a", "b", "c"]
+    df[1:] = [10, 11, 12]
```

Comment:
Is this because pandas does no longer upcast the dtype?

Reply:
Yes, and since the goal of the test was to test the slice, I figured I would just change the values that way.

Otherwise in 2.1, we get a `FutureWarning`